build/cloudbuild_kube-rbac-proxy.yaml

```diff
@@ -16,9 +16,9 @@ substitutions:
   # This is the kube-rbac-proxy version, source image tags for which must exist remotely.
   _KUBE_RBAC_PROXY_VERSION: v0.14.1
 steps:
-- name: "gcr.io/cloud-builders/docker"
+- name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211118-2f2d816b90'
   env:
   - "KUBE_RBAC_PROXY_VERSION=${_KUBE_RBAC_PROXY_VERSION}"
   entrypoint: "/usr/bin/env"
   args: ["bash", "-c", "./build/build.sh"]
-images: ["gcr.io/kubebuilder/kube-rbac-proxy:${_KUBE_RBAC_PROXY_VERSION}"]
+images: ["gcr.io/k8s-staging-kubebuilder/kube-rbac-proxy:${_KUBE_RBAC_PROXY_VERSION}"]
```

Comment:
Hi @rikatz,

See that I just changed the step name and image to be built.
Is that right? We need to:
- a) build gcr.io/k8s-staging-kubebuilder/kube-rbac-proxy that will be triggered by:
- b) add the image built in https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-kubebuilder/images.yaml so that it will be available via the registry, and we can download it with `docker pull k8s.io/kubebuilder/kube-rbac-proxy:<TAG>`

c/c @varshaprasad96

Reply:
We use the image to re-tag it see `gcr.io/kubebuilder/kube-rbac-proxy` the above one. So, I don't know how we would use it anymore. Maybe I am missing something. 

See that we download quay.io/brancz/kube-rbac-proxy from https://github.com/brancz/kube-rbac-proxy and then rebuild it with a new tag so that we can store that.

Reply:
The `images` field is not required but can be useful to explicitly declare the name of the image(s) you want to build.